cmd/collect_signals/collector/registry.go

```diff
@@ -0,0 +1,128 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ossf/criticality_score/cmd/collect_signals/projectrepo"
+	"github.com/ossf/criticality_score/cmd/collect_signals/signal"
+)
+
+var globalRegistry = NewRegistry()
+
+type Registry struct {
+	cs []*collectorWrapper
+}
+
+// NewRegistry creates a new instance of Registry.
+func NewRegistry() *Registry {
+	return &Registry{cs: []*collectorWrapper{}}
+}
+
+// containsCollector returns true if c has already been registered.
+func (r *Registry) containsCollector(c Collector) bool {
+	for _, cw := range r.cs {
+		if cw.Collector == c {
+			return true
+		}
+	}
+	return false
+}
+
+// Register adds the Collector c to the registry to be used when Collect is
+// called.
+//
+// This method may panic if the Collector's signal Set is not valid, or if the
+// Collector has already been added.
+func (r *Registry) Register(c Collector) {
+	validateCollector(c)
+	if r.containsCollector(c) {
+		panic(fmt.Sprintf("collector %s has already been registered", c.EmptySet().Namespace()))
+	}
+	if err := signal.ValidateSet(c.EmptySet()); err != nil {
+		panic(err)
+	}
+	r.cs = append(r.cs, &collectorWrapper{Collector: c})
+}
+
+func (r *Registry) collectorsForRepository(repo projectrepo.Repo) []Collector {
+	// Check for duplicates using a map to preserve the insertion order
+	// of the collectors.
+	exists := make(map[signal.Namespace]struct{})
+	res := make([]Collector, 0)
```

Comment:
nit: per https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices I think the preferred way to do this is `var res []Collector`?

Reply:
Done.